Components/Bites/src/OgreApplicationContextQt.cpp

```diff
@@ -131,6 +144,23 @@ namespace OgreBites
 
         p.miscParams["externalWindowHandle"] = std::to_string(size_t(window->winId()));
 
+#ifdef OGRE_WAYLAND
+
+        QPlatformNativeInterface *nativeInterface = QGuiApplication::platformNativeInterface();
+
+        if (!nativeInterface){
+          Ogre::LogManager::getSingleton().logMessage("[Qt] Native interface is nullptr!");
+        }
+
+        auto display = static_cast<wl_display*>(nativeInterface->nativeResourceForWindow("display", nullptr));
+        auto surface = static_cast<wl_surface*>(nativeInterface->nativeResourceForWindow("surface", window));
+
+        // Is window (QWindow) actually an instance of wl_egl_window?
+        p.miscParams["externalWindowHandle"] = Ogre::StringConverter::toString(size_t(window));
```

Comment:
are display and surface not sufficient for integration as with SDL?

Reply:
Yes it is! I will remove it. There are still other issues with this implementation (and not yet functional), though.